src/shell.js

```diff
@@ -421,6 +410,19 @@ if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
 #endif // ASSERTIONS
 }
 
+// --pre-jses are emitted after the JS helpers, and the Module
+// integration code, so they can reference locateFile and
+// the ENVIRONMENT_IS_* helpers, as well as the Module itself
+// (or even overwrite the entire Module).
+// {{PRE_JSES}}
+
+// Sometimes an existing Module object exists with properties
+// meant to overwrite the default module functionality. Here
+// we collect those properties and reapply _after_ we configure
+// the current environment's defaults to avoid having to be so
+// defensive during initialization.
+var moduleOverrides = Object.assign({}, Module);
```

Comment:
AFAIK, this should be safe, at least for that `Module.locateFile` in pre-js use-case (I'm using that without issues [here](https://github.com/kleisauke/wasm-vips/blob/9f6cc94a2a2d116cc6c75aaedc7f57793d74eb00/src/locatefile-cors-pre.js)). For example, the `locateFile` helper for the Wasm binary is used `{pre,post}amble.js`, which are emitted after this shell file.

I'll check if this is really safe for the other handlers that can be overridden on the incoming module, this may indeed cause issues if it's referenced/used before this line.